lms/static/js/dashboard/track_events.js

```diff
@@ -8,75 +8,101 @@ var edx = edx || {};
     'use strict';
 
     edx.dashboard = edx.dashboard || {};
-
+console.log('track_events.js');
     // Generate the properties object to be passed along with business intelligence events.
     edx.dashboard.generateTrackProperties = function(element){
         var $el = $(element),
             properties = {};
 
         properties.category = 'dashboard';
-        properties.label = $el.data("course-key");
+        properties.label = $el.data('course-key');
 
         return properties;
     };
 
-    edx.dashboard.TrackEvents = function() {
+    // Generate opbject to be passed with programs events
+    edx.dashboard.generateProgramProperties = function(element) {
+        var $el = $(element);
 
-         var course_title_link = $(".course-title > a"),
-            course_image_link = $(".cover"),
-            enter_course_link = $(".enter-course"),
-            options_dropdown = $(".wrapper-action-more"),
-            course_learn_verified = $(".verified-info"),
-            find_courses_btn = $(".btn-find-courses");
+        return {
+            category: 'dashboard',
+            course_id: $el.closest('.course-container').find('.info-course-id').html(),
+            program_id: $el.data('program-id')
```

Comment:
Are these custom properties (course_id and program_id) usable by the parties consuming this data?